apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/app/BlockchainProcessorImpl.java

```diff
@@ -1472,7 +1472,9 @@ private void scan(int height, boolean validate, boolean shutdown) {
                             if (validate) {
                                 blockEvent.select(literal(BlockEventType.BLOCK_SCANNED), new AnnotationLiteral<ScanValidate>() {}).fire(currentBlock);
                             } else {
-                                blockEvent.select(literal(BlockEventType.BLOCK_SCANNED)).fire(currentBlock);
+                                if (shardInitialHeight == 0 || currentBlock.getHeight() >= (shardInitialHeight + trimService.getMaxRollback())){
```

Comment:
I think, that its not the best solution. Such event should be triggered independently from sharding configuration

Reply:
If you have better solution, please describe it here. So we can discuss and decide.